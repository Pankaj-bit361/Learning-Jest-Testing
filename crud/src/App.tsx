import React from 'react';
import logo from './logo.svg';
import './App.css';
import TodoInput from './components/TodoInput';
import Todo from './components/Todo';

function App() {
  return (
    <div className="App">
<Todo/>
    </div>
  );
}

export default App;
