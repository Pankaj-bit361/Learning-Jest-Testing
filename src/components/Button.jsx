import React from 'react'
import styles from "./Button.module.css"

const Button = ({children,color,size,func}) => {
  return (
    <div>
      <button data-testid="custom-button" className={`${styles[color]} ${styles[size]}`}
      onClick={func}
      >{children}</button>
    
    </div>
  )
}

export default Button
