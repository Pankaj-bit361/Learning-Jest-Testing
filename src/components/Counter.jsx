import React, { useState } from 'react'

const Counter = () => {
const [count,setcount]=useState(0)

const handleadd=()=>{
    setcount((prev)=>prev+1)
}
  return (
    <div>
      <h2 data-testid="counter">Counter:{count}</h2>
      <button data-testid="add-button" onClick={handleadd}>ADD</button>
    </div>
  )
}

export default Counter
