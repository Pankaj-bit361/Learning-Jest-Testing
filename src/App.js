
import './App.css';
import Button from './components/Button';
import Counter from './components/Counter';

function App() {
  return (
    <div className="App">
     <Button size={"large"} color={"teal"} >Click Me</Button>
     <Button size={"medium"} color={"red"}>Check it</Button>
     <Counter/>
    </div>
  );
}

export default App;
