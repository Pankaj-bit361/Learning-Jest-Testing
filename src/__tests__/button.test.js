import Button from "../components/Button"
import {fireEvent, render,screen} from "@testing-library/react"
import App from "../App";
import Counter from "../components/Counter";
import renderer from "react-test-renderer"
;

describe("Testing button Component",()=>{
    it("Button Should Render",()=>{
render(<Button color={"teal"} size={"large"}>Click Me</Button>)

screen.debug()
const button=screen.getByText("Click Me")
expect(button).toBeInTheDocument();
    });
it("Buttom should be present in dom",()=>{
    render(<Button>Custom Button</Button>);
    const button=screen.getAllByTestId("custom-button")
    expect(button[0]).toBeInTheDocument("")
})
  it("Button should have text ",()=>{
    render(<App/>)
    const buttons=screen.getAllByTestId("custom-button")
expect(buttons[0]).toHaveTextContent("Click Me")
expect(buttons[1]).toHaveTextContent("Check it")
  }) 

  it("should invoke the function",()=>{
    const mockfunc=jest.fn()
    render(<Button func={mockfunc}/>);
    const button=screen.getByTestId("custom-button")
    fireEvent.click(button);

    expect(mockfunc).toBeCalledTimes(1)

  })


//Snapshot Testing

it("Should have snapshot",()=>{
    const tree=renderer.create(<Button color={"teal"} size={"large"}>Click Me</Button>).toJSON()
    expect(tree).toMatchSnapshot()
})

})



//for Counter

