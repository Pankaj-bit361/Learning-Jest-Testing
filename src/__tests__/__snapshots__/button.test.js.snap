// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button Component Should have snapshot 1`] = `
<div>
  <button
    className="teal large"
    data-testid="custom-button"
  >
    Click Me
  </button>
</div>
`;
